### DISCUSSION TITLE
API Create Rule: Only business workers can create orders for that business

### DISCUSSION THREAD

**Discussion by lbensaad** [pocketbase/pocketbase#5723] [Q&A]
Hi, I have these tables:
```
user(id, person_id)
person(id)
biz(id)
biz_worker(id,biz_id, person_id)
order(id,party_id)
party(id,biz_id1 (seller), biz_id2 (buyer))
```
how to ensure that only users that work in a business can list,view, and create orders of that business (as a seller or as a buyer).

**Comment by NeuWasFlaggedAgain** [comment]
I tried for a little while to think about a solution to this. While this probably can be expressed by PB rules, I'd instead use a custom go/js hook. The relations/ACLs you're after sound pretty complicated.

If you want, I can take another look and try to write my own hook when I get to a desktop computer!

**Comment by ararrocks** [comment]
in parse i would create an ACL Group named "Business1" and then add that user to that group. Each order then gets that ACL_group as a read/write. 

there is probably something similar here too?

**Comment by dschissler** [comment]
You'll probably want to use a `?=` operator to do a table join.

https://pocketbase.io/docs/api-rules-and-filters/
